js/auth.js

```diff
@@ -0,0 +1,306 @@
+
+  if (
+    isNicknameEmpty ||
+    isEmailEmpty ||
+    isPasswordEmpty ||
+    isPasswordCheckEmpty ||
+    NicknameError ||
+    hasEmailError ||
+    hasPasswordError ||
+    hasPasswordCheckError
+  ) {
+    signupButton.disabled = true;
+  } else {
+    signupButton.disabled = false;
+  }
+};
+
+const LoginButtonClick = (e) => {
+  e.preventDefault();
+  if (!loginButton.disabled) {
+    console.log("로그인 시도:", loginEmail.value);
+    window.location.href = "../pages/items.html";
+  }
+};
+
+const SignupButtonClick = (e) => {
+  e.preventDefault();
+  if (!signupButton.disabled) {
+    console.log("회원가입 시도:", signupEmail.value);
+    window.location.href = "../pages/login.html";
+  }
+};
+
+// const LoginPasswordToggleClick = () => {
+//   const passwordInput = document.getElementById("loginPassword");
+//   const toggleImg = passwordToggle.querySelector("img");
+
+//   if (passwordInput.type === "password") {
+//     passwordInput.type = "text";
+//     toggleImg.src = "../img/eye_on.svg";
+//     passwordToggle.classList.add("show");
+//   } else {
+//     passwordInput.type = "password";
+//     toggleImg.src = "../img/eye_off.svg";
+//     passwordToggle.classList.remove("show");
+//   }
+// };
+
+// const SignupPasswordToggleClick = () => {
+//   const passwordInput = document.getElementById("signupPassword");
+//   const toggleImg = signupPasswordToggle.querySelector("img");
+
+//   if (passwordInput.type === "password") {
+//     passwordInput.type = "text";
+//     toggleImg.src = "../img/eye_on.svg";
+//     signupPasswordToggle.classList.add("show");
+//   } else {
+//     passwordInput.type = "password";
+//     toggleImg.src = "../img/eye_off.svg";
+//     signupPasswordToggle.classList.remove("show");
+//   }
+// };
+
+// const SignupPasswordCheckToggleClick = () => {
+//   const passwordInput = document.getElementById("signupPasswordCheck");
+//   const toggleImg = passwordCheckToggle.querySelector("img");
+
+//   if (passwordInput.type === "password") {
+//     passwordInput.type = "text";
+//     toggleImg.src = "../img/eye_on.svg";
+//     passwordCheckToggle.classList.add("show");
+//   } else {
+//     passwordInput.type = "password";
+//     toggleImg.src = "../img/eye_off.svg";
+//     passwordCheckToggle.classList.remove("show");
+//   }
+// };
+
+const PasswordToggleClick = (inputId, toggleId) => {
+  const passwordInput = document.getElementById(inputId);
+  const toggleImg = document.getElementById(toggleId).querySelector("img");
+
+  if (passwordInput.type === "password") {
+    passwordInput.type = "text";
+    toggleImg.src = "../img/eye_on.svg";
+    document.getElementById(toggleId).classList.add("show");
+  } else {
+    passwordInput.type = "password";
+    toggleImg.src = "../img/eye_off.svg";
+    document.getElementById(toggleId).classList.remove("show");
+  }
+};
+
+// 로그인 비밀번호 토글 버튼 이벤트
+if (passwordToggle) {
+  passwordToggle.addEventListener(
```

Comment:
💬 여담
변수가 null 또는 undefined일 때만 안전하게 넘어가도록 하려면 optional chaining을 활용하는 것도 좋은 방법입니다.
이렇게 작성하면 존재 여부를 따로 조건문으로 확인하지 않아도 돼서 코드가 조금 더 간결해집니다.

[mdn](https://developer.mozilla.org/ko/docs/Web/JavaScript/Reference/Operators/Optional_chaining)

```suggestion
  passwordToggle?.addEventListener(
```